nixos/tests/systemd-nspawn-options.nix

```diff
@@ -0,0 +1,48 @@
+# Test for NixOS container options.
+
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "systemd-nspawn-options";
+
+  machine =
+    { pkgs, ... }:
+    {
+      systemd.nspawn.test-container = {
+        environment = { IN_CONTAINER = "1"; };
+        bind = { "/bind-dest" = { source = "/bind-src"; }; "/tmp" = {}; };
+        bindReadOnly = { "/nix/store" = { source = "/nix/store"; mountOption = "norbind"; }; };
+        temporaryFileSystem = [ "/tmpfs" { path = "/var/tmp"; mountOption = "mode=700"; } ];
+        inaccessible = [ "/home" "/root" ];
+        overlay = { "/var" = [ "+/var" "" ]; };
+        overlayReadOnly = { "/lib64" = [ "/lib" "/usr/lib64" ]; };
+        virtualEthernetExtra = { "veth-container" = "veth-host"; };
+        execConfig = { Parameters = "/bin/sh"; };
+        filesConfig = { ReadOnly = false; };
+        networkConfig = { Private = false; };
+      };
+      systemd.disableUnprivilegedNspawn = false;
+    };
+
+  testScript =
+    ''
+      $machine->waitForUnit("default.target");
+      $machine->succeed("find /etc/systemd/nspawn -follow");
+      my $conf = $machine->succeed("cat /etc/systemd/nspawn/test-container.nspawn");
+
+      $conf =~ /Environment="IN_CONTAINER=1"/ or die;
+      $conf =~ /Parameters=\/bin\/sh/ or die;
+
+      $conf =~ /BindReadOnly=\/nix\/store:\/nix\/store:norbind/ or die;
+      $conf =~ /Bind=\/bind-src:\/bind-dest/ or die;
+      $conf =~ /Bind=\/tmp/ or die;
+      $conf =~ /TemporaryFileSystem=\/tmpfs/ or die;
+      $conf =~ /TemporaryFileSystem=\/var\/tmp:mode=700/ or die;
+      $conf =~ /Inaccessible=\/home/ or die;
+      $conf =~ /Inaccessible=\/root/ or die;
+      $conf =~ /OverlayReadOnly=\/lib:\/usr\/lib64:\/lib64/ or die;
+      $conf =~ /Overlay=\+\/var::\/var/ or die;
+      $conf =~ /ReadOnly=false/ or die;
```

Comment:
As we boot into a VM here, those tests usually serve as integration tests, so I'd prefer to see some kind of test here which actually checks if nspawn machines properly start up (for inspiration you might have a look at `nixos/tests/containers-*.nix`).